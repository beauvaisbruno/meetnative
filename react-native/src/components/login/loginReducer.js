"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const reducerHelper_1 = require("../../utils/reducerHelper");
var LoginActions;
(function(LoginActions) {
  LoginActions["loginGoogle"] = "loginGoogle";
  LoginActions["loginFacebook"] = "loginFacebook";
  LoginActions["loginWithEmail"] = "loginWithEmail";
  LoginActions["loginLoadUser"] = "loginLoadUser";
  LoginActions["loginResetEmail"] = "loginResetEmail";
})((LoginActions = exports.LoginActions || (exports.LoginActions = {})));
const initialState = {
  loginGoogle: new reducerHelper_1.ActionResult(),
  loginFacebook: new reducerHelper_1.ActionResult(),
  loginWithEmail: new reducerHelper_1.ActionResult(),
  loginLoadUser: new reducerHelper_1.ActionResult(),
  loginResetEmail: new reducerHelper_1.ActionResult()
};
function login(state = initialState, action) {
  state = reducerHelper_1.reduceAction(LoginActions, action, state);
  return state;
}
exports.login = login;
