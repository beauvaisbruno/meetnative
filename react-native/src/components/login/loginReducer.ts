import {
  ActionResult,
  IActionResult,
  reduceAction
} from "../../utils/reducerHelper";

export enum LoginActions {
  loginGoogle = "loginGoogle",
  loginFacebook = "loginFacebook",
  loginWithEmail = "loginWithEmail",
  loginLoadUser = "loginLoadUser",
  loginResetEmail = "loginResetEmail"
}
export interface ILoginState {
  readonly loginGoogle: IActionResult<any>;
  readonly loginFacebook: IActionResult<any>;
  readonly loginWithEmail: IActionResult<any>;
  readonly loginLoadUser: IActionResult<any>;
  readonly loginResetEmail: IActionResult<undefined>;
}

const initialState: ILoginState = {
  loginGoogle: new ActionResult<any>(),
  loginFacebook: new ActionResult<any>(),
  loginWithEmail: new ActionResult<any>(),
  loginLoadUser: new ActionResult<any>(),
  loginResetEmail: new ActionResult<undefined>()
};

export function login(state = initialState, action: any) {
  state = reduceAction(LoginActions, action, state);
  return state;
}
