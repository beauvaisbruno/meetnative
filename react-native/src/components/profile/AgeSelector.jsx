"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var native_base_1 = require("native-base");
var lang_1 = require("./lang");
var React = require("react");
var custom_1 = require("../../config/native-base-theme/variables/custom");
var createRows = function () {
    var rows = [];
    for (var i = 18; i < 77; i++) {
        rows.push(
        //@ts-ignore
        <native_base_1.Picker.Item label={lang_1.default("{age} yo", { age: i })} value={i} key={i}/>);
    }
    return rows;
};
exports.AgeSelector = function (_a) {
    var value = _a.value, onChange = _a.onChange;
    return (<native_base_1.Picker testID={"picker"} mode="dropdown" iosIcon={<native_base_1.Icon name="arrow-down"/>} placeholder={lang_1.default("Select your age")} placeholderStyle={{ color: custom_1.default.textGrey }} placeholderIconColor={custom_1.default.textGrey} style={{ width: undefined }} selectedValue={value} onValueChange={function (age) {
        onChange(age);
    }}>
      {createRows()}
    </native_base_1.Picker>);
};
