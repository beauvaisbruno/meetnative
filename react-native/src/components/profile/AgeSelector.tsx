import { Icon, Picker } from "native-base";
import lang from "./lang";
import * as React from "react";
import custom from "../../config/native-base-theme/variables/custom";

const createRows = () => {
  const rows = [];
  for (let i = 18; i < 77; i++) {
    rows.push(
      //@ts-ignore
      <Picker.Item label={lang("{age} yo", { age: i })} value={i} key={i} />
    );
  }
  return rows;
};

export const AgeSelector = ({
  value,
  onChange
}: {
  value: number;
  onChange: (age: number) => void;
}) => {
  return (
    <Picker
      testID={"picker"}
      mode="dropdown"
      iosIcon={<Icon name="arrow-down" />}
      placeholder={lang("Select your age")}
      placeholderStyle={{ color: custom.textGrey }}
      placeholderIconColor={custom.textGrey}
      style={{ width: undefined }}
      selectedValue={value}
      onValueChange={(age: number) => {
        onChange(age);
      }}
    >
      {createRows()}
    </Picker>
  );
};
