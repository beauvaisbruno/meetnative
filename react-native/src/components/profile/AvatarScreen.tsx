import { NavigationInjectedProps, withNavigation } from "react-navigation";
import * as React from "react";
import { Content, View } from "native-base";
import { avatars } from "./img/Avatars";
import { TouchableHighlight } from "react-native";
import AutoHeightImage from "react-native-auto-height-image";
import { widthPercentageToDP as wp } from "react-native-responsive-screen";

export const AvatarScreen = ({ navigation }: NavigationInjectedProps) => {
  const rows: JSX.Element[] = [];
  let even = true;
  let cols: JSX.Element[] = [];
  Object.keys(avatars).forEach(index => {
    cols.push(
      <TouchableHighlight
        onPress={() => {
          navigation.navigate("ProfileContainer", { avatar: index });
        }}
        key={index}
        testID={index}
      >
        <AutoHeightImage source={avatars[index] as number} width={wp("50%")} />
      </TouchableHighlight>
    );
    if (!even) {
      rows.push(
        <View key={index} style={{ flexDirection: "row" }}>
          {cols}
        </View>
      );
      cols = [];
    }
    even = !even;
  });

  return <Content>{rows}</Content>;
};

export default withNavigation(AvatarScreen);
