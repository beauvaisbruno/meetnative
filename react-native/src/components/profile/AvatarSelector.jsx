"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var native_base_1 = require("native-base");
var React = require("react");
var react_native_1 = require("react-native");
var Avatars_1 = require("./img/Avatars");
var react_native_responsive_screen_1 = require("react-native-responsive-screen");
var react_native_auto_height_image_1 = require("react-native-auto-height-image");
var react_navigation_1 = require("react-navigation");
var AvatarSelector = /** @class */ (function (_super) {
    __extends(AvatarSelector, _super);
    function AvatarSelector() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    AvatarSelector.prototype.componentWillUnmount = function () {
        if (this.navListener)
            this.navListener.remove();
    };
    AvatarSelector.prototype.componentDidMount = function () {
        var _this = this;
        this.navListener = this.props.navigation.addListener("willFocus", function (payload) {
            if (_this.props.navigation.state.params &&
                _this.props.navigation.state.params.avatar) {
                _this.props.onChange(_this.props.navigation.state.params.avatar);
                _this.props.navigation.state.params.avatar = undefined;
            }
        });
    };
    AvatarSelector.prototype.render = function () {
        var _this = this;
        return (<native_base_1.View style={{
            textAlignVertical: "top",
            justifyContent: "center",
            alignItems: "center"
        }}>
        <native_base_1.View style={{ textAlignVertical: "top", flex: 1, flexDirection: "row" }}>
          <native_base_1.Text style={{ textAlignVertical: "top", marginRight: 10 }}>
            Avatar
          </native_base_1.Text>
          <react_native_1.TouchableHighlight testID={"TouchableHighlight"} onPress={function () {
            _this.props.navigation.navigate("AvatarScreen");
        }}>
            <react_native_auto_height_image_1.default source={Avatars_1.avatars[this.props.value]} width={react_native_responsive_screen_1.widthPercentageToDP("45%")}/>
          </react_native_1.TouchableHighlight>
        </native_base_1.View>
      </native_base_1.View>);
    };
    return AvatarSelector;
}(React.Component));
exports.AvatarSelector = AvatarSelector;
exports.default = react_navigation_1.withNavigation(AvatarSelector);
