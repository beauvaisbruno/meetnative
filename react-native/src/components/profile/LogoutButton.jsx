"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = require("react");
var native_base_1 = require("native-base");
var react_redux_1 = require("react-redux");
var profileReducer_1 = require("./profileReducer");
var reducerHelper_1 = require("../../utils/reducerHelper");
var lang_1 = require("./lang");
var LogoutButton = /** @class */ (function (_super) {
    __extends(LogoutButton, _super);
    function LogoutButton() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    LogoutButton.prototype.render = function () {
        var _this = this;
        return (<native_base_1.View padder>
        <native_base_1.Button testID={"logoutButton"} block rounded primary onPress={function () { return _this.props.doLogout(); }}>
          <native_base_1.Icon name="logout" type="AntDesign" color="white" fontSize={30}/>
          <native_base_1.Text>{lang_1.default("Logout")}</native_base_1.Text>
        </native_base_1.Button>
      </native_base_1.View>);
    };
    return LogoutButton;
}(react_1.Component));
exports.LogoutButton = LogoutButton;
exports.mapStateToProps = function (state) {
    return __assign({}, state.profile.profileLogout);
};
exports.mapDispatchToProps = function (dispatch) {
    return {
        doLogout: function () { return dispatch(reducerHelper_1.fetchingAction(profileReducer_1.ProfileActions.profileLogout)); }
    };
};
exports.default = react_redux_1.connect(exports.mapStateToProps, exports.mapDispatchToProps)(LogoutButton);
