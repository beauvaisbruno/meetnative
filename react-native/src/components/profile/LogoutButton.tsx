import React, { Component } from "react";
import { Button, Text, View, Icon } from "native-base";
import { connect } from "react-redux";
import { ProfileActions } from "./profileReducer";
import { Dispatch } from "redux";
import { IApplicationState } from "../../rootReducers";
import { fetchingAction, IActionResult } from "../../utils/reducerHelper";
import lang from "./lang";

interface IProps {
  doLogout: () => void;
}

export class LogoutButton extends Component<IProps & IActionResult<undefined>> {
  render() {
    return (
      <View padder>
        <Button testID={"logoutButton"} block rounded primary onPress={() => this.props.doLogout()}>
          <Icon name="logout" type="AntDesign" color="white" fontSize={30} />
          <Text>{lang("Logout")}</Text>
        </Button>
      </View>
    );
  }
}

export const mapStateToProps = (state: IApplicationState) => {
  return { ...state.profile.profileLogout };
};

export const mapDispatchToProps = (dispatch: Dispatch) => {
  return {
    doLogout: () => dispatch(fetchingAction(ProfileActions.profileLogout))
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(LogoutButton);
