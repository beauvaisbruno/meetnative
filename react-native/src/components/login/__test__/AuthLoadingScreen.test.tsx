import { shallow } from "enzyme";

import { IApplicationState } from "../../../rootReducers";
import * as reducerHelper from "../../../utils/reducerHelper";
import { FormikActions } from "formik";
import { ILoginState, LoginActions } from "../loginReducer";
import * as React from "react";
import AuthLoadingScreenContainer, { AuthLoadingScreen,
  mapDispatchToProps, mapStateToProps } from "../AuthLoadingScreen";
import { LogoutButton } from "../../profile/LogoutButton";
import toJson from "enzyme-to-json";

it("Container", () => {
  // @ts-ignore
  const wrapper = shallow(<AuthLoadingScreenContainer/>);
  expect(wrapper).toBeDefined();
  expect(mapStateToProps({ loginLoadUser: { expected: "expected" } } as unknown as ILoginState))
    .toEqual({ expected: "expected" });

  const dispatchMock = jest.fn();
  const fetchingAction = jest.fn().mockReturnValue("fetchingAction");
  jest.spyOn(reducerHelper, "fetchingAction")
    .mockImplementation(fetchingAction);

  const map = mapDispatchToProps(dispatchMock);
  map.loadUser();
  expect(dispatchMock).lastCalledWith("fetchingAction");
  expect(fetchingAction).lastCalledWith(LoginActions.loginLoadUser);
});

it("Component" , () => {
  const loadUser = jest.fn();
  const tree = shallow(
    <AuthLoadingScreen
      loadUser={loadUser}
      {...new reducerHelper.ActionResult<undefined>()}
    />
  );
  expect(toJson(tree)).toMatchSnapshot();
  expect(loadUser).toBeCalled();
});
