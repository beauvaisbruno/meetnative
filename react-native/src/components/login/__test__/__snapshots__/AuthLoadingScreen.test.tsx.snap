// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component 1`] = `
<View>
  <Loading />
  <StatusBar
    animated={false}
    barStyle="default"
    showHideTransition="fade"
  />
</View>
`;
